scipy/spatial/transform/_rotation.pyx

```diff
@@ -1865,6 +1873,118 @@ cdef class Rotation:
+            be positive.
+        modulus : None
+            This overridden argument is not applicable to Rotations and must be
+            None.
+
+        Returns
+        -------
+        composition : `Rotation` instance
+            If the input Rotation ``p`` contains ``N`` multiple rotations, then
+            the output will contain ``N`` rotations where the ``i``th rotation
+            is equal to ``p[i] ** n``
+
+        Notes
+        -----
+        For example, a power of 2 will double the angle of rotation, and a
+        power of 0.5 will halve the angle. There are two notable cases: if
+        ``n == 1`` then the original rotation is returned, and if ``n == 0``
+        then the identity rotation is returned.
+
+        Note that fractional powers ``0 < n < 1`` which effectively take a root
+        of rotation, do so using the shortest path smallest representation of
+        that angle (the principal root). This means that powers of ``n`` and
+        ``1/n`` are not necessarily inverses of each other. For example, a 0.5
+        power of a +240 degree rotation will be calculated as the 0.5 power of
+        a -120 degree rotation, with the result being a rotation of -60 rather
+        than +120 degrees.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Raising a rotation to a power:
+
+        >>> p = R.from_rotvec([1, 0, 0])
+        >>> q = p ** 2
+        >>> q.as_rotvec()
+        array([2., 0., 0.])
+        >>> r = p ** 0.5
+        >>> r.as_rotvec()
+        array([0.5, 0., 0.])
+
+        Inverse powers do not necessarily cancel out:
+
+        >>> p = R.from_rotvec([0, 0, 120], degrees=True)
+        >>> ((p ** 2) ** 0.5).as_rotvec(degrees=True)
+        array([  -0.,   -0., -60.])
+        """
+        cdef Py_ssize_t num_rotations = len(self._quat)
+        cdef double angle, scale, angle2, n3, n5
+        cdef double[:, :] result = _empty2(num_rotations, 4)
+        cdef double[:] quat
+
+        if modulus is not None:
+            raise NotImplementedError("modulus not supported")
+
+        if n < 0:
+            raise ValueError("Rotations cannot be raised to negative powers.")
+
+        # short-cuts
+        elif n == 0:
+            return Rotation.identity(num_rotations)
+
+        elif n == 1:
+            result = self._quat.copy()
+
+        elif n == 2:
+            result = _compose_quat(self._quat, self._quat)
+
+        # general scaling of rotation angle
+        else:
+            for ind in range(num_rotations):
+                if self._quat[ind, 3] < 0:  # w > 0 to ensure 0 <= angle <= pi
+                    quat = self._quat[ind, :].copy()
+                    for i in range(4):
+                        quat[i] *= -1
+                else:
+                    quat = self._quat[ind, :]
+                angle = 2 * atan2(_norm3(quat), abs(quat[3]))
+
+                if angle <= 1e-3:  # small angle Taylor series expansion
```

Comment:
Here's another idea for this: 

Do we know about what value for `norm(quat[:3])` the zero division starts to create numerical problems?

An interesting problem could be to analyze how the error between `sin(n*angle/2) / sin(angle/2)` changes with the number of terms of the Taylor series, according to the distance to `0`, and then use a smaller `eps` (smaller than `1e-3`) so that we can simply state `sin(n*angle/2) / sin(angle/2) = n`

Reply:
I just did a quick test. Defined these functions:
```
def f(theta,n):
    return np.sin(n*theta/2) / np.sin(theta/2)

def g(theta,n):

    n3 = n*n*n
    n5 = n*n*n3
    return n + (n - n3) / 24 * theta + (3 * n5 - 10 * n3 + 7 * n) / 5760 * theta * theta
```
And then did some tests with different values:
```
f(1e-10, 100) == 100
g(1e-10, 100) == 99.99999583375006
f(1e-10, 1000) == 999.9999999999997
g(1e-10, 1000) == 999.9958333427084
```
We can see that the original function `f` gives values closer to the `n`. So I really think that lowering the `eps` value for the angle and then reducing the Taylor series to a simple `n` can actually give a more accurate result, apart from been a lil bit faster and a lot cleaner.

Reply:
Alternatively, if there is a Cython implementation of the `sinc` function (if there is, I'll suppose it's defined as the numpy version, which is `sinc(x) = sin(pi*x) / (pi*x)`), then for small `angle`:
```
sin(n * angle / 2) / sin(angle / 2) 
    == sin(n * angle / 2) / (angle / 2) 
    == n * sin(n * angle / 2) / (n * angle / 2)
```
Which gives:
```
sin(n * angle / 2) / sin(angle / 2) == n * sinc(n * angle / (2 * pi))
```
This would probably be better, since it would lower the error caused by large values of `n` when taking only the linear term of the sine Taylor expansion.

Reply:
FYI: The function $$f(x, n) = \frac{\sin(n x / 2)}{n \sin(x/2)}$$ is known as the periodic sinc function or the Dirichlet function.  We have an implementation at [`scipy.special.diric`](https://scipy.github.io/devdocs/reference/generated/scipy.special.diric.html).  I haven't checked if that provides any advantage over the code in this PR--the factor of `n`  in `n*diric(angle, n)` might multiply the numerical error in an undesirable way.  (Also note that `diric` requires `n > 0`, so an additional line or two of code would be needed to handle `n <= 0`.)

Reply:
Interesting, this is even better! 
In this PR's implementation, we can safely assume that `n > 0` by this stage.

For the numerical error, I think it would be better, but alternatively an extra case could be added:
```
        if angle <= 1e-7 and n <= 1e-3:
                    scale = n
        elif angle <= 1e-7:
                    scale = n * diric(angle, n)
        else:  # large angle
                    scale = sin(n * angle / 2) / sin(angle / 2)
```

Reply:
That's an interesting look @evbernardes, thanks for running that! There are two other taylor series expansions in this file, they would probably benefit from another PR with the same sort of look.

Reply:
Alright, just checked it. For the first one:
```
sin(angle / 2) / angle == sinc(angle / (2 * pi)) / 2
```
Or, using `diric` as pointed out by @WarrenWeckesser :
```
sin(angle / 2) / angle == diric(angle, 1) / 2
```
I think you could just replace everything by that instead of having the case checking.

Alternatively, since here we don't have the problem of having the `n` factor that can also vary, I think in this case you can just use a the smaller `eps` and and use the first term of the Taylor expansion for small angles: `sinc(angle / 2) / angle == 1/2`.

Reply:
For the other case, it's basically the reciprocal of that... 
So either remove case checking and:
```
angle / sin(angle / 2)  == 2 / sinc(angle / (2 * pi))
angle / sin(angle / 2)  == 2 / diric(angle, 1)
```
Or keep case checking, lower `eps` and for small values:
```
angle / sin(angle / 2)  == 2
```
I haven't seen the code much, but I imagine there are no other singularity problems with dividing by sine, since `angle/2` only grows up to 90 degrees, right?

Reply:
I updated the approximation introduced in this PR, but I think changes to the other two should be broken out separately. I opened this issue to track it: https://github.com/scipy/scipy/issues/17635